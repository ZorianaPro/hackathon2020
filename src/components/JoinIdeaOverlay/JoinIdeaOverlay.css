.JoinIdeaOverlay .JoinIdeaOverlay-Title {
    font-size: var(--fontSize56);
    font-weight: var(--fontWeightPoppinsBold);
    font-style: italic;
    line-height: var(--lineHeight);
    text-align: center;
    color: var(--colorSpaceCadet);
}