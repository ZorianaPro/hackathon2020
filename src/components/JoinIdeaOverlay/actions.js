export const actions = {
	open: 'OPEN:JOINIDEA:OVERLAY',
	close: 'CLOSE:JOINIDEA:OVERLAY',
	submit: 'SUBMIT:JOINIDEA:OVERLAY',
	error: 'ERROR:JOINIDEA:OVERLAY'
};

export const open = (content = {}) => ({
	type: actions.open,
	value: content
});

export const close = () => ({
	type: actions.close,
});

export const submit = content => ({
	type: actions.submit,
	value: content
});

export const error = content => ({
	type: actions.error,
	value: content
});

export default {
	open,
	close,
	submit,
	error
};
