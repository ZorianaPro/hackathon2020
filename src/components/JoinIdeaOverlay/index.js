import React, { useState, useEffect, useCallback } from 'react';
import { useSelector, useDispatch } from "react-redux";
import './JoinIdeaOverlay.css';
import '../Overlay/Overlay.css';
import { close, submit } from './actions';

const JoinIdeaOverlay = ({
	team,
  dispatch,
	isOpen,
	ideaId
}) => {

	dispatch = useDispatch();

	const [firstName, setFirstName] = useState('');
	const [lastName, setLastName] = useState('');
	const [email, setEmail] = useState('');
	const [position, setPosition] = useState('');
	const [submitFrom, setSubmitForm] = useState(false);
	const [showThankYou, setShowThankYou] = useState(false);

	[isOpen, ideaId] = useSelector(state => [
		state.joinIdeaOverlayReducer.open,
		state.joinIdeaOverlayReducer.content.id
	]);

	const handleChange = useCallback(
		(target) => {
			const field = target.name;
			const value = target.value;
			switch (field) {
				case 'firstName':
					return setFirstName(value);
				case 'lastName':
					return setLastName(value);
				case 'email':
					return setEmail(value);
				case 'position':
					return setPosition(value);
			}

		}, []
	);

	const closeOverlay = useCallback(
		() => {
			dispatch(close())
		}, []
	);

	const joinIdea = useCallback(
		(e) => {
			e.preventDefault();
			setSubmitForm(true);
			return false
		}, []
	);

	useEffect(() => {

		if (submitFrom) {
			const data = {
				ideaId: ideaId,
				firstName: firstName,
				lastName: lastName,
				email: email,
				position: position
			};
			console.log(data);
			// do all this in promise and clean state after submit
			dispatch(submit(data));
			setSubmitForm(false);
			setShowThankYou(true);
			setTimeout(() => {
				dispatch(close());
				setShowThankYou(false);
			}, 3000)
		} else {

		}
	}, [submitFrom]);

	const animateLabel = useCallback(
		(input, focus=true) => {
		const labelName = input.name;
		const labels = document.getElementsByTagName('label');
			for( let i = 0; i < labels.length; i++ ) {
				if (labels[i].htmlFor === labelName)
					if (!focus) {
						if (input.value === '') {
							labels[i].classList.remove('focus');
						}
					} else {
						labels[i].classList.add('focus');
					}
			}
		},[]);

	return (
		<div className="JoinIdeaOverlay">
			{ isOpen
				&& <div className="Overlay">
					<div className="Overlay-Container">
						{ !showThankYou
							&& <div className="Overlay-Box">
							<div className="Overlay-Close" onClick={() => closeOverlay()}>
								close
							</div>
							<div className="JoinIdeaOverlay-Title">
								Join the idea
							</div>
							<form  className="JoinIdeaOverlay-Form" action="" onSubmit={(e) => joinIdea(e)} method="POST">
								<div className="JoinIdeaOverlay-Input-Container">
									<label htmlFor="firstName" className={`${firstName !== '' ? 'focus' : ''}`}>First Name*</label>
									<input type="text" name="firstName" required value={firstName}
									       onFocus={(e) => animateLabel(e.target, true)}
									       onBlur={(e) => animateLabel(e.target, false)}
									       onChange={(e) => handleChange(e.target)}/>
								</div>
								<div className="JoinIdeaOverlay-Input-Container">
									<label htmlFor="lastName" className={`${lastName !== '' ? 'focus' : ''}`}>Last Name*</label>
									<input type="text" name="lastName" required value={lastName}
									       onFocus={(e) => animateLabel(e.target, true)}
									       onBlur={(e) => animateLabel(e.target, false)}
									       onChange={(e) => handleChange(e.target)}/>
								</div>
								<div className="JoinIdeaOverlay-Input-Container">
									<label htmlFor="email" className={`${email !== '' ? 'focus' : ''}`}>Your email*</label>
									<input type="email" name="email" required value={email}
									       onFocus={(e) => animateLabel(e.target, true)}
									       onBlur={(e) => animateLabel(e.target, false)}
									       onChange={(e) => handleChange(e.target)}/>
								</div>
								<div className="JoinIdeaOverlay-Input-Container">
									<label htmlFor="position" className={`${position !== '' ? 'focus' : ''}`}>Position*</label>
									<input type="text" name="position" required value={position}
									       onFocus={(e) => animateLabel(e.target, true)}
									       onBlur={(e) => animateLabel(e.target, false)}
									       onChange={(e) => handleChange(e.target)}/>
								</div>
								<button type="submit">Submit</button>
							</form>
						</div>
						}
						{	showThankYou
							&& <div className="Overlay-Box">
							<div className="Overlay-Close" onClick={() => closeOverlay()}>
								close
							</div>
							<div>Thank you</div>
						</div>
						}
						</div>
					</div>
			}
		</div>
	);
};

export default JoinIdeaOverlay;