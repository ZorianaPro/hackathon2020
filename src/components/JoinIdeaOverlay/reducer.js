import { actions } from "./actions";

const initialState = {
	open: false,
	content: {},
	error: '',
	userInfo: {}
};

export default (state = initialState, action) => {
	switch (action.type) {
		case actions.open:
			return {
				...state,
				open: true,
				content: action.value
			};
		case actions.close:
			return {
				...state,
				open: false
			};
		case actions.submit:
			return {
				...state,
				userInfo: action.value
			};
		case actions.error:
			return {
				...state,
				error: action.value
			};
		default:
			return state;
	}
};
