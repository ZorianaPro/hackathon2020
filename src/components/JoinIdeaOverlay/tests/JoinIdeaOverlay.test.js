import React from 'react';
import { shallow, mount } from 'enzyme';
import JoinIdeaOverlay from '../';

describe('JoinIdeaOverlay', () => {

  it('renders without crashing', () => {
    expect(shallow.bind(shallow, <JoinIdeaOverlay />)).not.toThrow();
  });

  it('renders as expected', () => {
    expect(mount(<JoinIdeaOverlay />)).toMatchSnapshot();
  });
});
