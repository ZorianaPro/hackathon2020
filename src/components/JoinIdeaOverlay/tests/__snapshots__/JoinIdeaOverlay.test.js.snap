// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinIdeaOverlay renders as expected 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <JoinIdeaOverlay>
    <Overlay
      onClose={[Function]}
    >
      <div
        className="Overlay Overlay--isOpening"
      >
        <div
          className="Overlay-Content"
          style={
            Object {
              "animationDuration": "200ms",
            }
          }
        >
          <div
            className="Overlay-Mask"
            onClick={[Function]}
          />
          <div
            className="Overlay-Box"
          >
            <div
              className="Overlay-Close"
              onClick={[Function]}
            >
              close
            </div>
            <div
              className="JoinIdeaOverlay"
            >
              <div
                className="JoinIdeaOverlay-Title"
              >
                Join the idea
              </div>
              <form
                method="POST"
                onSubmit={[Function]}
              >
                <div>
                  <label
                    className=""
                    htmlFor="firstName"
                  >
                    First Name*
                  </label>
                  <input
                    name="firstName"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="text"
                  />
                </div>
                <div>
                  <label
                    className=""
                    htmlFor="lastName"
                  >
                    ast Name*
                  </label>
                  <input
                    name="lastName"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="text"
                    value=""
                  />
                </div>
                <div>
                  <label
                    className=""
                    htmlFor="email"
                  >
                    Your email*
                  </label>
                  <input
                    name="email"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="email"
                  />
                </div>
                <div>
                  <label
                    className=""
                    htmlFor="position"
                  >
                    Position*
                  </label>
                  <input
                    name="position"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="text"
                  />
                </div>
                <button
                  className="Button Button--isSecondaryTransparent"
                  type="submit"
                >
                  <p
                    className="Button-Inside"
                  >
                    Submit
                  </p>
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </Overlay>
  </JoinIdeaOverlay>
</Provider>
`;
